### PR TITLE
highlights(php): fix queries related to namespaces

### PR DESCRIPTION
**Pull request by zweimach**
This PR could have been a discussion thread, but I wanted to show the code.

This PR refines the queries related to namespaces and scoping.
Mainly for consistency with other languages' namespace-related highlights.

| Before | After |
| ------ | ----- |
| image | image |
| image | image |
| image | image |

Comparison with other languages:

| Language | Screenshot |
| -------- | ---------- |
| PHP (after this PR) | image |
| Rust | image |
| C++ | image |
| Elm | image |

While at it, I also fixed queries related to `@constant`, `@function.call`, and captures for the `as` keyword:
| Before | After |
| ------ | ----- |
| image | image |
| image | image |
| image | image |

I will tag everyone involved in my previous PR: @tk-shirasaka @Lucario387 @gbprod @theHamsta.
What do you think?

### CODE REVIEW

File: queries/php/highlights.scm [hunk 1]
```diff
@@ -2,39 +2,55 @@
 
 (variable_name) @variable
 
+; Constants
+
+((name) @constant
+ (#vim-match? @constant "^_?[A-Z][A-Z\d_]*$"))
+((name) @constant.builtin
+ (#vim-match? @constant.builtin "^__[A-Z][A-Z\d_]+__$"))
```
Comment (theHamsta) [inline queries/php/highlights.scm:10]: looks like this could also be handled by the faster lua match?

**Review by theHamsta — APPROVED** [review]

File: queries/php/highlights.scm [hunk 2]
```diff
@@ -2,39 +2,55 @@
...
 (enum_declaration
   name: (name) @type)
 (interface_declaration
   name: (name) @type)
 (namespace_use_clause
-  [(name) (qualified_name)] @type)
-(namespace_aliasing_clause (name)) @type.definition
+  [(name) @type
+   (qualified_name (name) @type)])
+(namespace_aliasing_clause (name) @type.definition)
```
Comment (theHamsta) [inline queries/php/highlights.scm:35]: wasn't aware that we actually use this weird capture

File: queries/php/highlights.scm (hunk 1 above)
Comment (zweimach) [reply queries/php/highlights.scm]: Thanks, I have added your suggestion.

**Comment by gbprod** [comment]
Not sure it's interesting here but I really like the way vscode hightlight php use : 

2023-01-30-155147_439x51_scrot

That's why I add in my custom highlights this code to have the same colors: 

```scheme
(namespace_name_as_prefix) @namespace
```

2023-01-30-155416_527x300_scrot

2023-01-30-155430_468x199_scrot

Do you think it could be the default behavior ?

**Comment by lucario387** [comment]
> Do you think it could be the default behavior ?

This make it clearer to the end user, I really like it, just like how semantic tokens in Jdtls allow me to somewhat know whether I import classes, or annotations or other node types (of course its not necessary, but syntax sugar is beautiful/looks better no?)

image

**Comment by tk-shirasaka** [comment]
I think it is generally good, but I was curious about the handling of aliases.
In other languages, aliases seem to be assigned the same highlight as the origin, so I thought it would be better to follow this practice.

| Language  | Screenshot | Screenshot (use alias) |
|:------------:|:-------------:|:------------------------:|
| PHP           | image | image |
| Python      | image | image |
| Typescript | image | image |

**Comment by zweimach** [comment]
I actually agree with @tk-shirasaka.
@theHamsta What do you think? Does `@type.definition` includes import renaming too?
But that is out of the scope of this PR. It should be in a separate PR.

**Review by theHamsta — APPROVED** [review]

**Pull request merged**